cmd/vtcatomicswap/main.go

```diff
@@ -0,0 +1,1010 @@
+func getRawChangeAddress(c *rpc.Client) (vtcutil.Address, error) {
+	rawResp, err := c.RawRequest("getrawchangeaddress", nil)
+	if err != nil {
+		return nil, err
+	}
+	var addrStr string
+	err = json.Unmarshal(rawResp, &addrStr)
+	if err != nil {
+		return nil, err
+	}
+	addr, err := vtcutil.DecodeAddress(addrStr, chainParams)
+	if err != nil {
+		return nil, err
+	}
+	if !addr.IsForNet(chainParams) {
+		return nil, fmt.Errorf("address %v is not intended for use on %v",
+			addrStr, chainParams.Name)
+	}
+	return addr, nil
+}
+
+func promptPublishTx(c *rpc.Client, tx *wire.MsgTx, name string) error {
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		fmt.Printf("Publish %s transaction? [y/N] ", name)
+		answer, err := reader.ReadString('\n')
+		if err != nil {
+			return err
+		}
+		answer = strings.TrimSpace(strings.ToLower(answer))
+
+		switch answer {
+		case "y", "yes":
+		case "n", "no", "":
+			return nil
+		default:
+			fmt.Println("please answer y or n")
+			continue
+		}
+
+		txHash, err := c.SendRawTransaction(tx, false)
+		if err != nil {
+			return fmt.Errorf("sendrawtransaction: %v", err)
+		}
+		fmt.Printf("Published %s transaction (%v)\n", name, txHash)
+		return nil
+	}
+}
+
+// contractArgs specifies the common parameters used to create the initiator's
+// and participant's contract.
+type contractArgs struct {
+	them       *vtcutil.AddressPubKeyHash
+	amount     vtcutil.Amount
+	locktime   int64
+	secretHash []byte
+}
+
+// builtContract houses the details regarding a contract and the contract
+// payment transaction, as well as the transaction to perform a refund.
+type builtContract struct {
+	contract       []byte
+	contractP2SH   vtcutil.Address
+	contractTxHash *chainhash.Hash
+	contractTx     *wire.MsgTx
+	contractFee    vtcutil.Amount
+	refundTx       *wire.MsgTx
+	refundFee      vtcutil.Amount
+}
+
+// buildContract creates a contract for the parameters specified in args, using
+// wallet RPC to generate an internal address to redeem the refund and to sign
+// the payment to the contract transaction.
+func buildContract(c *rpc.Client, args *contractArgs) (*builtContract, error) {
+	refundAddr, err := getRawChangeAddress(c)
+	if err != nil {
+		return nil, fmt.Errorf("getrawchangeaddress: %v", err)
+	}
+	refundAddrH, ok := refundAddr.(interface {
+		Hash160() *[ripemd160.Size]byte
+	})
+	if !ok {
+		return nil, errors.New("unable to create hash160 from change address")
+	}
+
+	contract, err := atomicSwapContract(refundAddrH.Hash160(), args.them.Hash160(),
+		args.locktime, args.secretHash)
+	if err != nil {
+		return nil, err
+	}
+	contractP2SH, err := vtcutil.NewAddressScriptHash(contract, chainParams)
+	if err != nil {
+		return nil, err
+	}
+	contractP2SHPkScript, err := txscript.PayToAddrScript(contractP2SH)
+	if err != nil {
+		return nil, err
+	}
+
+	feePerKb, minFeePerKb, err := getFeePerKb(c)
+	if err != nil {
+		return nil, err
+	}
+
+	unsignedContract := wire.NewMsgTx(2)
```

Comment:
Should this be version 1 or 2?  Looking through vertcoin's transaction.h, version 1 is the current version while 2 is the max standard version, implying that version 2 transactions are't ready to send yet.

https://github.com/vertcoin/vertcoin/blob/ec464a06f35da791c829b6f745b35d2ed9ca5a8b/src/primitives/transaction.h#L306

Reply:
We've been using version 2 transactions for LN and SegWit. If/when we release a new Core update that can send to/receive such transactions natively we'll bump the default transaction version there too (e.g. https://vtc.xchan.gr/blockExplorer/tx/3d0ba63444e75c7d751a8f2ba081e44cac60d0cabb2c0bb1bfa7ec1bfa40b717).

I assume using version 1 there would be fine then given that this code doesn't use segwit.